src/server/endpoints/covidcast_utils/model.py

```diff
@@ -272,3 +283,128 @@ def map_row(source: str, signal: str) -> str:
         return signal_source.source
 
     return transformed_pairs, map_row
+
+
+def _resolve_all_signals(
+    source_signals: Union[SourceSignalPair, List[SourceSignalPair]], data_sources_by_id: DataFrame
+) -> Union[SourceSignalPair, List[SourceSignalPair]]:
+    if isinstance(source_signals, SourceSignalPair):
+        if source_signals.signal == True:
+            source = data_sources_by_id.get(source_signals.source)
+            if source:
+                return SourceSignalPair(source.source, [s.signal for s in source.signals])
+        return source_signals
+    if isinstance(source_signals, list):
+        return [_resolve_all_signals(pair, data_sources_by_id) for pair in source_signals]
+    raise TypeError("source_signals is not Union[SourceSignalPair, List[SourceSignalPair]].")
+
+
+def _buffer_and_tag_iterator(
+    iter: Iterable[Dict], name_dict: Dict[Any, List[Any]], keyfunc: Callable[[Dict], Any]
```

Comment:
consider using generics instead of the Any to better indicate the connection between the different arguments